src/libs/sfdk/packagesignkeys.cpp

```diff
@@ -0,0 +1,238 @@
+#include "packagesignkeys.h"
+#include "sfdkconstants.h"
+#include "sfdkglobal.h"
+
+#include <app/app_version.h>
+#include <utils/algorithm.h>
+#include <utils/environment.h>
+#include <utils/fileutils.h>
+#include <utils/qtcprocess.h>
+#include <QDir>
+#include <QFileInfo>
+#include <QProcess>
+#include <QRegularExpression>
+#include <QSettings>
+
+#include <sfdk/buildengine.h>
+#include <sfdk/sdk.h>
+
+using Utils::Environment;
+using Utils::FilePath;
+using Utils::FilePaths;
+
+namespace {
+const char GPG_NOT_FOUND_ERROR[] = "PackageSign: Gpg not found.";
+}
+
+namespace Sfdk {
+
+class GpgRunner

```

Comment:
This needs to be implemented a non blocking way. Check e.g. how docker or vboxmanage is used in {docker,vbox}virtualmachine.cpp.

Reply:
Why is this needed? Why can't we do with a non-blocking completion wait in QEventLoop?
For few gpg calls looks like a excess mechanism.

Reply:
Sfdk API is meant to be 100% asynchronous - no blocking operation that could freeze the UI inadvertently. This is well supported internally, so the implementation overhead is too negligible to bother with the question whether this particular external command may or may not ever block our application. Use of nested event loops on library side is a asking for issues and should be avoided when possible.